src/workspaces/NewWorkspaceModal/NewWorkspaceWizard.tsx

```diff
@@ -0,0 +1,798 @@
+          padding: '-1 -1 rem',
+          width: '90%',
+          position: 'absolute',
+          bottom: '1rem',
+          ...modalStyles.buttonRow,
+          display: 'flex',
+          justifyContent: 'space-between',
+        }}
+      >
+        <ButtonSecondary style={{ position: 'absolute', left: '1rem' }} onClick={onDismiss}>
+          Cancel
+        </ButtonSecondary>
+        <div style={{ display: 'flex', gap: '1rem', marginLeft: 'auto' }}>
+          <ButtonSecondary
+            style={{ padding: '0 1rem', marginLeft: '1rem', border: '1px solid', borderRadius: '5px' }}
+            disabled={activeTab === 'basic' && errors}
+            tooltip={getTooltip()}
+            onClick={handleSecondaryButtonClick}
+          >
+            {Utils.cond([activeTab === 'basic', () => 'Quick create workspace'], () => 'Previous')}
+          </ButtonSecondary>
+          <ButtonPrimary
+            style={{ padding: '0 1rem', marginLeft: '1rem' }}
+            disabled={errors}
+            tooltip={activeTab === 'security' ? Utils.summarizeErrors(errors) : undefined}
+            onClick={handlePrimaryButtonClick}
+          >
+            {Utils.cond([activeTab === 'security', () => 'Create workspace'], () => 'Next')}
+          </ButtonPrimary>
+        </div>
+      </div>
+    );
+
+    return Utils.cond(
+      [loading, () => spinnerOverlay],
+      [
+        hasBillingProjects,
+        () => (
+          <Modal
+            title={Utils.cond(
+              [!!title, () => title],
+              [!!cloneWorkspace && creating, () => 'Cloning workspace'],
+              [creating, () => 'Creating workspace'],
+              [!!cloneWorkspace, () => 'Clone Workspace'],
+              () => 'Create a New Workspace'
+            )}
+            // Hold modal open while waiting for create workspace request.
+            shouldCloseOnOverlayClick={!creating}
+            shouldCloseOnEsc={!creating}
+            showButtons={false}
+            onDismiss={onDismiss}
+            okButton={
+              <ButtonPrimary disabled={errors} tooltip={Utils.summarizeErrors(errors)} onClick={create}>
+                {Utils.cond(
+                  [!!buttonText, () => buttonText],
+                  [!!cloneWorkspace, () => 'Clone Workspace'],
+                  () => 'Create Workspace'
+                )}
+              </ButtonPrimary>
+            }
+            width={620}
+            styles={{ modal: { height: 675 } }}
+          >
+            {creating ? (
+              <CreatingWorkspaceMessage />
+            ) : (
+              <>
+                <TabWizard
+                  aria-label='create workspace wizard'
+                  activeTab={activeTab}
+                  tabNames={['basic', 'sharing', 'security']}
+                  tabsDisabled={[false, activeTab === 'basic' && !!errors, activeTab === 'basic' && !!errors]}
+                  displayNames={{
+                    basic: '1. Basic Information and Billing',
```

Comment:
If you add numbers as keys for the tab, you could remove the numbers here and the display names could auto update with numbers added.